File: src/server/TypeLibrary.Api/Attributes/AttributeView.cs
using Tyle.Core.Attributes;
using Tyle.Core.Common;

namespace Tyle.Api.Attributes;

public class AttributeView : ImfType
{
    public RdlPredicate? Predicate { get; set; }
    public ICollection<RdlUnit> Units { get; set; } = new List<RdlUnit>();
    public int UnitMinCount { get; set; }
    public int UnitMaxCount { get; set; }
    public ProvenanceQualifier? ProvenanceQualifier { get; set; }
    public RangeQualifier? RangeQualifier { get; set; }
    public RegularityQualifier? RegularityQualifier { get; set; }
    public ScopeQualifier? ScopeQualifier { get; set; }
    public ValueConstraintView? ValueConstraint { get; set; }
}

File: src/client/src/common/types/attributes/attributeView.ts
import { ImfType } from "../common/imfType";
import { ProvenanceQualifier } from "./provenanceQualifier";
import { RangeQualifier } from "./rangeQualifier";
import { RdlPredicate } from "./rdlPredicate";
import { RdlUnit } from "./rdlUnit";
import { RegularityQualifier } from "./regularityQualifier";
import { ScopeQualifier } from "./scopeQualifier";
import { ValueConstraintView } from "./valueConstraintView";

export interface AttributeView extends ImfType {
  predicate: RdlPredicate | undefined;
  units: RdlUnit[];
  unitMinCount: number;
  unitMaxCount: number;
  provenanceQualifier: ProvenanceQualifier | undefined;
  rangeQualifier: RangeQualifier | undefined;
  regularityQualifier: RegularityQualifier | undefined;
  scopeQualifier: ScopeQualifier | undefined;
  valueConstraint: ValueConstraintView | undefined;
}
